File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/repositories/CategoryTagRepository.java
package com.decrux.pocketr.api.repositories;

import com.decrux.pocketr.api.entities.db.ledger.CategoryTag;
import java.util.List;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface CategoryTagRepository extends JpaRepository<CategoryTag, UUID> {

    List<CategoryTag> findByOwnerUserId(long userId);

    boolean existsByOwnerUserIdAndNameIgnoreCase(long userId, String name);
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr_api/repositories/CategoryTagRepository.kt
package com.decrux.pocketr_api.repositories

import com.decrux.pocketr_api.entities.db.ledger.CategoryTag
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface CategoryTagRepository : JpaRepository<CategoryTag, UUID> {

    fun findByOwnerUserId(userId: Long): List<CategoryTag>

    fun existsByOwnerUserIdAndNameIgnoreCase(userId: Long, name: String): Boolean
}
